### PR TITLE
Move match item workflow into MatchStageService

### PR DESCRIPTION
**Pull request by dardenkyle**
## Summary

- Move per-match fetch, parse, persist, follow-up queueing, and lifecycle updates into `MatchStageService`
- Wire `MatchController` to delegate per-item match processing to the service
- Keep controller ownership of batch coordination, retry policy, scraper reset/rotation, delays, and summary logging
- Add focused `MatchStageService` tests for success and parser-empty outcomes
- Update affected controller tests to patch the service dependency directly

## Validation

- `python -m pytest`

### CODE REVIEW

File: cs2_analytics/controllers/match_controller.py [hunk 1]
```diff
@@ -58,25 +67,14 @@ def run(self, batch_size: int = 25) -> None:
...
-                            soup, match_url
-                        )
-
-                        if match:
-                            store_matches([match])
-                            self._queue_followups(map_links, demo_links)
-                            self.match_queue.mark_as_processed(match_id)
+                        self.stage_service.scraper = scraper
+                        if self.stage_service.process_item(match_id, match_url):
                             succeeded += 1
```
[the remark this answers is not in the document]
Comment (dardenkyle) [reply cs2_analytics/controllers/match_controller.py:76]: I moved `mark_as_processing(match_id)` back into `MatchController` before the retry loop so it runs once per item, while `MatchStageService.process_item()` now handles only the per-attempt fetch/parse/persist/final state transitions. Focused match controller and match stage service tests are passing.

File: cs2_analytics/controllers/match_controller.py [hunk 2]
```diff
@@ -58,25 +67,14 @@ def run(self, batch_size: int = 25) -> None:
...
-                        match, map_links, demo_links = self.parser.parse_match(
-                            soup, match_url
-                        )
-
-                        if match:
-                            store_matches([match])
-                            self._queue_followups(map_links, demo_links)
-                            self.match_queue.mark_as_processed(match_id)
+                        self.stage_service.scraper = scraper
+                        if self.stage_service.process_item(match_id, match_url):
```
[the remark this answers is not in the document]
Comment (dardenkyle) [reply cs2_analytics/controllers/match_controller.py]: I updated this so MatchStageService no longer stores or mutates scraper state. The controller now keeps the active scraper local to the retry/rotation loop and passes it explicitly into process_item(..., scraper=scraper) for each attempt.

I also updated the stage-service tests to cover per-attempt scraper usage and ran the focused controller/stage-service test suite plus ruff/mypy checks.
